cmd/crossplane/core/core.go

```diff
@@ -59,7 +69,35 @@ func (c *Command) Run(log logging.Logger) error {
 		return errors.Wrap(err, "Cannot get config")
 	}
 
-	mgr, err := ctrl.NewManager(cfg, ctrl.Options{SyncPeriod: &c.Sync})
+	// approach 1:
+	// abstract away `ctrl.NewManager` call in crossplane-runtime
+	// ----------------------------------------------------------
+	// mgr, err := controller.NewManager(cfg, controller.Options{
+	// 	Options: &ctrl.Options{
+	// 		LeaderElection:          c.LeaderElection,
+	// 		LeaderElectionNamespace: c.Namespace,
+	// 		SyncPeriod:              &c.Sync,
+	// 	},
+	// 	ManagerName:      c.Name,
+	// 	ManagerNamespace: c.Namespace,
+	// })
+
+	// approach 2:
+	// keep calling `ctrl.NewManager` with additional config here
+	// ----------------------------------------------------------
+	if c.Name == "" {
+		return errors.New(errManagerName)
+	}
+	if c.Namespace == "" {
+		return errors.New(errManagerNamespace)
+	}
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		LeaderElection:             c.LeaderElection,
+		LeaderElectionID:           fmt.Sprintf("crossplane-leader-election-%s", c.Name),
+		LeaderElectionNamespace:    c.Namespace,
+		LeaderElectionResourceLock: resourcelock.LeasesResourceLock,
```

Comment:
I recall that we discussed wanting to use lease based locks so that folks, but I didn't realise until now that they're only available in controller-runtime 0.7.0, which is still in alpha. Given that we don't want to upgrade to 0.7.0 until it's out of alpha, I think we should just go with ConfigMap based locks for now.

Reply:
Yes, I agree. What I can suggest is either of the following options:

1) keep this PR as is until controller-runtime graduates out of alpha and then update dependencies and merge this
2) revert back to v0.6.0 and merge with this PR and in another one, prior to v0.14.0 release update to v0.7.0

I prefer the latter depending on the release time of controller-runtime, but generally prefer to directly go to Lease API in Crossplane v0.14.0, if possible.

Reply:
In the meantime, I can open another PR (similar to https://github.com/crossplane/crossplane-runtime/pull/210) to update `controller-runtime` to latest v0.7.0-alpha and fix potential issues, compilation errors, and such there so when the controller-runtime was released we can bring it over here as soon as possible (maybe even just in time for v0.14.0)